File: platforms/android/src/main/kotlin/co/trace/Trace.kt
package co.trace

// Umbrella module re-exports the public surface.
//
// Identity invariants (SPEC Implementation notes):
//   1. Hit-testing routes to the Rust core (TraceFile.hitTest), never to Skia.
//   2. Each addressable element gets its own canvas.drawPath call. SkPicture
//      caching is reserved for elements with `flags.isDecorative == true`.
//   3. The identity-to-handle map lives in TraceRenderer (Kotlin), not in
//      Skia's internals.

object Trace {
    const val FORMAT_MAJOR: Int = 1
    const val FORMAT_MINOR: Int = 0
}

File: platforms/ios/Sources/Trace/Trace.swift
// Trace.swift — umbrella module. Re-exports the public surface.
//
// The framework's public API is intentionally narrow per SPEC §5:
//   - TraceFile / TraceSession: read and stream `.trace` data
//   - TraceView: UIView that renders a Trace source via Skia
//   - TraceViewDelegate: callbacks for taps and animation completion
//   - HitMode: hit-test query mode
//
// Identity invariants (from SPEC's Implementation notes) hold across this
// framework:
//   1. Hit-testing routes to the Rust core's `hit_test`, never to Skia.
//   2. TraceRenderer draws each element with its own `canvas.drawPath`
//      (except `flags.is_decorative == true` elements, which may be batched
//      into a cached SkPicture).
//   3. The identity-to-handle map is owned in TraceRenderer (Swift), not by
//      Skia internals.

import Foundation

public enum Trace {
    public static let formatVersion: (major: UInt16, minor: UInt16) = (1, 0)
}
